tracing-subscriber/src/registry/mod.rs

```diff
@@ -0,0 +1,47 @@
+//! **EXPERIMENTAL**: Storage for span data shared by multiple [`Layer`]s.
+//!
+//! This module is experimental. Although potential breaking changes will be
+//! avoided when possible, we reserve the right to make breaking changes to this
+//! module until it is no longer experimental.
+//!
+//! Add the `registry_unstable` feature to your `Cargo.toml` to enable
+//! this module:
+//!
+//! ```toml
+//! [dependencies.tracing-subscriber]
+//! features = ["registry_unstable"]
+//! ```
+//!
+//! [`Layer`]: ../layer/struct.Layer.html
+use tracing_core::{span::Id, Metadata};
+/// Provides access to stored span metadata.
+///
+/// Subscribers which store span metadata and associate it with span IDs should
+/// implement this trait; if they do, any [`Layer`]s wrapping them can look up
+/// metadata via the [`Context`] type's [`metadata()`] method.
+///
+/// [`Layer`]: ../layer/struct.Layer.html
+/// [`Context`]: ../layer/struct.Context.html
+/// [`metadata()`]: ../layer/struct.Context.html#method.metadata
+pub trait LookupMetadata {
```

Comment:
will there be anyway to add custom data? This is something I do for all my layers.

Reply:
Yes, that'll be added in a future PR. It requires a little more work than just exposing metadata lookups.

I expect there will be separate traits for:
- Looking up metadata (added here)
- Looking up span relationships (e.g. given a span, iterate over its parents, children, followers, etc)
- Custom data (probably similar to the `http` crate's `Extensions`)